### PR TITLE
Build a sitemap

### PR DESCRIPTION
**Pull request by jvoros**
Build sitemap for Google Search console. Available at http://www.aliemcards.com/sitemap.txt

### CODE REVIEW

File: gulpfile.ts [hunk 1]
```diff
@@ -159,6 +160,22 @@ function writeFilePromise(path, content) {
   });
 }
 
+function sitemap(cards: SingleCardJSON[]): SingleCardJSON[] {
+  const baseurl = 'http://www.aliemcards.com';
```
Comment (dsifford) [inline gulpfile.ts]: `http` => `https`

File: gulpfile.ts [hunk 2]
```diff
@@ -159,6 +160,22 @@ function writeFilePromise(path, content) {
...
+  let sitemap = `${baseurl}\n`;
```
Comment (dsifford) [inline gulpfile.ts]: This is fine, but my preference would be to use a single, functional, immutable string, like so:

```js
const str = `
${baseurl}
${baseurl}/about
${baseurl}/categories
${baseurl}/cards
${cards.map(card => `${baseurl}/cards/${card.id}`).join('\n')}
`;
```

Your call

Comment (jvoros) [inline gulpfile.ts]: I like that better. Didn't think about using a function call within the string declaration. But I guess the `/`` is just a disguised function call.

Comment (jvoros) [inline gulpfile.ts]: The template literal caused ident problems. Stole your idea with map().join() and just built a single array for the sitemap.

**Pull request merged**